app/src/main/java/org/mozilla/fenix/settings/sitepermissions/SitePermissionsManagePhoneFeatureFragment.kt

```diff
@@ -73,66 +128,120 @@ class SitePermissionsManagePhoneFeatureFragment : Fragment() {
-        val expectedAction = if (phoneFeature == PhoneFeature.AUTOPLAY_AUDIBLE) BLOCKED else ASK_TO_ALLOW
-        radio.setOnClickListener {
-            saveActionInSettings(expectedAction)
+    }
+
+    private fun initFourthRadio(rootView: View) {
+        with(rootView.fourth_radio) {
+            if (phoneFeature == AUTOPLAY_AUDIBLE) {
+                visibility = View.VISIBLE
+                text = getCombinedLabel(
+                    getString(R.string.preference_option_autoplay_blocked2),
+                    getString(R.string.phone_feature_recommended)
+                )
+                setOnClickListener {
+                    saveActionInSettings(it.context, AUTOPLAY_BLOCK_ALL)
+                }
+                restoreState(AUTOPLAY_BLOCK_ALL)
+            } else {
+                visibility = View.GONE
+            }
         }
-        radio.restoreState(expectedAction)
     }
 
-    private fun RadioButton.restoreState(action: SitePermissionsRules.Action) {
-        if (phoneFeature.getAction(settings) == action) {
+    private fun RadioButton.restoreState(buttonAction: SitePermissionsRules.Action) {
+        if (phoneFeature.getAction(settings) == buttonAction) {
             this.isChecked = true
             this.setStartCheckedIndicator()
         }
     }
 
-    private fun initSecondRadio(rootView: View) {
-        val radio = rootView.findViewById<RadioButton>(R.id.block_radio)
-        radio.text = when (phoneFeature) {
-            PhoneFeature.AUTOPLAY_AUDIBLE, PhoneFeature.AUTOPLAY_INAUDIBLE ->
-                getString(R.string.preference_option_autoplay_allowed)
-            else -> getString(R.string.preference_option_phone_feature_blocked)
+    private fun RadioButton.restoreState(buttonAutoplaySetting: Int) {
+        if (settings.getAutoplayUserSetting(AUTOPLAY_BLOCK_ALL) == buttonAutoplaySetting) {
+            this.isChecked = true
+            this.setStartCheckedIndicator()
         }
-        val expectedAction = if (phoneFeature == PhoneFeature.AUTOPLAY_AUDIBLE) ALLOWED else BLOCKED
-        radio.setOnClickListener {
-            saveActionInSettings(expectedAction)
+    }
+
+    private fun saveActionInSettings(action: SitePermissionsRules.Action) {
+        settings.setSitePermissionsPhoneFeatureAction(phoneFeature, action)
+    }
+
+    /**
+     * Saves the user selected autoplay setting.
+     *
+     * See [Settings.setAutoplayUserSetting] kdoc for an explanation of why this cannot follow the
+     * same code path as other permissions.
+     */
+    private fun saveActionInSettings(context: Context, autoplaySetting: Int) {
+        settings.setAutoplayUserSetting(autoplaySetting)
+        val (audible, inaudible) = when (autoplaySetting) {
+            AUTOPLAY_ALLOW_ALL -> ALLOWED to ALLOWED
+            AUTOPLAY_ALLOW_ON_WIFI -> {
+                addWifiConnectedListener(context.application)
+                return
+            }
+            AUTOPLAY_BLOCK_AUDIBLE -> BLOCKED to ALLOWED
```

Comment:
This setting confuse me a bit.

 In the UI this is "Block audio", when the user selects this setting, I think he/she will be expected **just audio to be blocked** not video ,right? But What this setting is stating is `audible = BLOCKED` and `inaudible = ALLOWED`, this will cause that **videos that are not muted will be blocked** as they contain audible content, can you help me to clarify?

Reply:
Yeah, we're matching desktop behavior, but it is a little bit confusing.  @betsymi do you have any thoughts on this?  If we do want to make any changes, we'll have to reach out to desktop about their copy too.

Reply:
Is this the setting we're talking about, or someplace else? Could you provide a screenshot so I can see the full context of the string? 

![Screenshot_20200309-152229](https://user-images.githubusercontent.com/43555492/76254121-e857e400-6219-11ea-8710-32daa60ab822.png)

Reply:
@betsymi yep, that's the screen.  After this PR lands, it will look like this though:

https://user-images.githubusercontent.com/4400286/72918787-1f6e4680-3cfb-11ea-9520-bcdafe562618.png